gradle-plugin/src/test/kotlin/io/spine/protodata/gradle/ExtensionTest.kt

```diff
@@ -82,18 +85,23 @@ class `Plugin extension should` {
     }
 
     @Test
-    fun `add dependency tasks`() {
-        val taskName = "generateCustomProto"
-        extension.generateProtoTasks(taskName)
-        assertThat(extension.generateProtoTasks.get())
-            .contains(taskName)
+    fun `specify request file location`() {
+        val path = "/my/path/to/main.bin"
+        extension.requestFilesDir = path
+        assertThat(extension.requestFilesDirProperty.get().asFile)
+            .isEqualTo(project.projectDir.resolve(path))
     }
 
     @Test
-    fun `specify request file location`() {
-        val path = "/my/path/to/file.bin"
-        extension.requestFile = path
-        assertThat(extension.requestFileProperty.get().asFile)
-            .isEqualTo(project.projectDir.resolve(path))
+    fun `produce source directory`() {
+        val basePath = "my/path"
+        val subDir = "foobar"
+
+        extension.srcBaseDir = basePath
+        extension.srcSubDir = subDir
+
+        val sourceDir = extension.sourceDir(project.sourceSets.getByName(MAIN_SOURCE_SET_NAME))
+        assertThat(sourceDir.get().asFile.toPath())
+            .isEqualTo(project.projectDir.toPath() / basePath / MAIN_SOURCE_SET_NAME / subDir)
```

Comment:
This is fun!

Reply:
Inspired by Python, as far as I can tell.